### PR TITLE
NETOBSERV-1790: Manage enrichment via "k8s.v1.cni.cncf.io/network-status"

### PR DESCRIPTION
**Pull request by jotak**
## Description

Manage enrichment by extracting pod IPs from the annotation "k8s.v1.cni.cncf.io/network-status", which is used (at least) by multus

This allows to correlate Pods with their Mac / IPs on secondary interfaces

### Breaking change

The API for `add_kubernetes_infra` transform stage has been modified to directly use enriched name & namespace to figure out the layer, instead of doing an informer lookup by IP. The reason is informers can now index by MAC.

To migrate, if `add_kubernetes_infra` rule was used such as:

```yaml
    kubernetes_infra:
      inputs:
      - SrcAddr
      - DstAddr
      output: K8S_FlowLayer
```

it must be changed such as:

```yaml
    kubernetes_infra:
      namespaceNameFields:
      - namespace: SrcK8S_Namespace
        name: SrcK8S_Name
      - namespace: DstK8S_Namespace
        name: DstK8S_Name
      output: K8S_FlowLayer
```

where SrcK8S_Namespace, SrcK8S_Name, DstK8S_Namespace and DstK8S_Name match the names of the corresponding enriched fields.

## Dependencies

https://github.com/netobserv/network-observability-operator/pull/732 for Mac enrichment

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [x] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Review by jpinsonneau — APPROVED** [review]
LGTM, thanks @jotak !

**Comment by jpinsonneau** [comment]
https://github.com/netobserv/flowlogs-pipeline/pull/674/commits/f39388d718f3a6714b8103160e2e3d0f7d6cf7de
^ checking on mac addresses when possible

File: pkg/pipeline/transform/kubernetes/enrich.go [hunk 1]
```diff
@@ -17,12 +17,17 @@ func MockInformers() {
...
-	return informers.InitFromConfig(kubeConfigPath)
+func InitFromConfig(config api.NetworkTransformKubeConfig) error {
+	return informers.InitFromConfig(config)
 }
 
 func Enrich(outputEntry config.GenericMap, rule api.K8sRule) {
-	kubeInfo, err := informers.GetInfo(fmt.Sprintf("%s", outputEntry[rule.Input]))
+	ip := fmt.Sprintf("%s", outputEntry[rule.Input])
+	// TODO: replace this by a new rule input
+	mac := fmt.Sprintf("%s", outputEntry[strings.Replace(rule.Input, "Addr", "Mac", -1)])
```
Comment (jpinsonneau) [inline pkg/pipeline/transform/kubernetes/enrich.go:27]: This need to be changed before merge
I'm still discussing about putting this behind a feature gate or not at this stage

Comment (jpinsonneau) [reply pkg/pipeline/transform/kubernetes/enrich.go]: https://github.com/netobserv/flowlogs-pipeline/pull/674/commits/5c0f987bffdf1c59257291c36635e389809c7ad2

File: pkg/pipeline/transform/kubernetes/enrich.go [hunk 2]
```diff
@@ -21,10 +21,12 @@ func InitFromConfig(config api.NetworkTransformKubeConfig) error {
...
 }
 
-func Enrich(outputEntry config.GenericMap, rule api.K8sRule) {
+func Enrich(outputEntry config.GenericMap, rule *api.K8sRule) {
 	ip := fmt.Sprintf("%s", outputEntry[rule.Input])
-	// TODO: replace this by a new rule input
-	mac := fmt.Sprintf("%s", outputEntry[strings.Replace(rule.Input, "Addr", "Mac", -1)])
+	var mac string
+	if len(rule.MacInput) > 0 {
+		mac = fmt.Sprintf("%s", outputEntry[rule.MacInput])
```
Comment (jotak) [inline pkg/pipeline/transform/kubernetes/enrich.go:28]: For info, for performance we should avoid using `fmt.Sprintf("%s", ...)` whenever possible especially when it's part of the flows processing path - it's not only here but in a bunch of other places too. I've just created a task for that: https://issues.redhat.com/browse/NETOBSERV-1799

**Comment by jotak** [comment]
thanks @jpinsonneau !
/lgtm

**Review by jpinsonneau — APPROVED** [review]

**Comment by memodi** [comment]
it's working well to enrich traffic for secondary interfaces. 

/label qe-approved

**Review by OlivierCazade — APPROVED** [review]
LGTM

**Pull request merged**